src/ColumnsPaginatedBookView.ts

```diff
@@ -92,24 +96,46 @@ export default class ColumnsPaginatedBookView implements PaginatedBookView {
         body.style.width = "";
         body.style.marginLeft = "";
         body.style.marginRight = "";
+        body.style.marginTop = "";
+        body.style.marginBottom = "";
+        this.bookElement.contentDocument.documentElement.style.height = "";
         this.bookElement.style.height = "";
         this.bookElement.style.width = "";
     }
 
     /** Returns the total width of the columns that are currently
         positioned to the left of the iframe viewport. */
     private getLeftColumnsWidth(): number {
-        return -(this.bookElement.contentDocument.body.style.left || "0px").slice(0, -2);
+        const body = HTMLUtilities.findRequiredElement(this.bookElement.contentDocument, "body") as HTMLBodyElement;
+
+        const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') !== -1;
+        const isXML = this.bookElement.src.indexOf(".xml") !== -1;
+        if (isFirefox && isXML) {
+            // This handles books from Feedbooks with XML resources.
```

Comment:
A little more elaboration here would be nice. I don't understand the need for this special case.